pkg/storage/engine.go

```diff
@@ -539,28 +539,25 @@ type Writer interface {
 	//
 	// It is safe to modify the contents of the arguments after it returns.
 	ClearRawRange(start, end roachpb.Key) error
-	// ClearMVCCRangeAndIntents removes MVCC keys and intents from start (inclusive)
-	// to end (exclusive). This is a higher-level method that handles both
-	// interleaved and separated intents. Similar to the other Clear* methods,
-	// this method actually removes entries from the storage engine.
+	// ClearMVCCRange removes MVCC keys from start (inclusive) to end (exclusive),
+	// including intents. Similar to the other Clear* methods, this method
+	// actually removes entries from the storage engine.
 	//
 	// It is safe to modify the contents of the arguments after it returns.
-	ClearMVCCRangeAndIntents(start, end roachpb.Key) error
-	// ClearMVCCRange removes MVCC keys from start (inclusive) to end
-	// (exclusive). It should not be expected to clear intents, though may clear
-	// interleaved intents that it encounters. It is meant for efficiently
+	ClearMVCCRange(start, end roachpb.Key) error
+	// ClearMVCCVersions removes MVCC key versions from start (inclusive) to end
+	// (exclusive). It does not affect intents. It is meant for efficiently
 	// clearing a subset of versions of a key, since the parameters are MVCCKeys
 	// and not roachpb.Keys. Similar to the other Clear* methods, this method
 	// actually removes entries from the storage engine.
 	//
 	// It is safe to modify the contents of the arguments after it returns.
-	ClearMVCCRange(start, end MVCCKey) error
-
-	// ClearIterRange removes all keys in the given span using an iterator to
-	// iterate over point keys and remove them from the storage engine using
-	// per-key storage tombstones (not MVCC tombstones). Any separated
-	// intents/locks will also be cleared.
-	ClearIterRange(start, end roachpb.Key) error
+	ClearMVCCVersions(start, end MVCCKey) error
+	// ClearMVCCIteratorRange removes all keys in the given span using an MVCC
+	// iterator and clearing individual keys, including intents. Similar to the
```

Comment:
and clears?

Do we need to clarify what is the difference between interator clear and range clear since we are changing comments here? Is the difference that ClearRange would put a pebble range tombstone while iterator is always putting point tombstones during deletion?

Reply:
Yep, `ClearMVCCIteratorRange` uses Pebble point tombstones and `ClearMVCCRange` use Pebble range tombstones. I've spelled this out on all the range clears.

>and clears?

I think the -ing form is right here? Anyway, I reworded it to remove any doubt, it was a bit clumsily worded anyway.